slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -451,6 +451,12 @@ std::string SLAMServiceImpl::GetNextDataFileOnline() {
     while (b_continue_session) {
         const auto file_list_online =
             viam::io::ListSortedFilesInDirectory(path_to_data);
+        if (delete_processed_data && first_processed_file_index >= 0) {
+            for (int i = first_processed_file_index;
```

Comment:
[q] Correct me if I'm wrong but `first_processed_file_index` is being used to ensure that we delete all files AFTER the one that was previously found by `GetNextDataFile`, including said file.

Shouldn't this be all the files before this one? Or is that to protect previous runs of data?

Reply:
This is to protect previous runs of data.

Reply:
I'd argue that ListSortedFilesInDirectory should only return current runs of data

Reply:
I like this idea - or at least a variation of it. I'll try that out.

Reply:
Created a ticket to fix this: https://viam.atlassian.net/browse/RSDK-1659

Reason is that I want this bug fix to go in for the rest of the team.